### PR TITLE
enarx-keep-sgx-shim: link rust musl compiler intrinsics to the shim

### PR DESCRIPTION
**Pull request by haraldh**
Otherwise memcpy() and friends would have to be implemented.
Remove obsolete enarx-keep-sgx-shim/src/libc.rs.

### CODE REVIEW

**Comment by haraldh** [comment]
```console
[haraldh@coffeelake enarx-keep-sgx-shim]$ CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_RUNNER="../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim " cargo run
   Compiling enarx-keep-sgx-shim v0.1.0 (/home/haraldh/git/enarx/enarx-keep-sgx-shim)
    Finished dev [unoptimized + debuginfo] target(s) in 0.26s
     Running `/home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/debug/enarx-keep-sgx-shim`
arg: "/init"
env: "LANG=C"
aux: ExecFilename("/init")
aux: Platform("x86_64")
aux: Uid(1000)
aux: EUid(1000)
aux: Gid(1000)
aux: EGid(1000)
aux: PageSize(4096)
aux: Secure(false)
aux: ClockTick(100)
aux: Flags(0)
aux: HwCap(0)
aux: HwCap2(0)
aux: PHdr(35184376283200)
aux: PHent(56)
aux: PHnum(10)
aux: Random([195, 87, 208, 31, 104, 11, 46, 132, 80, 20, 20, 16, 68, 47, 50, 104])
[haraldh@coffeelake enarx-keep-sgx-shim]$ CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_RUNNER="../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim " cargo run --release
   Compiling enarx-keep-sgx-shim v0.1.0 (/home/haraldh/git/enarx/enarx-keep-sgx-shim)
    Finished release [optimized] target(s) in 2.77s
     Running `/home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim`
arg: "/init"
env: "LANG=C"
aux: ExecFilename("/init")
aux: Platform("x86_64")
aux: Uid(1000)
aux: EUid(1000)
aux: Gid(1000)
aux: EGid(1000)
aux: PageSize(4096)
aux: Secure(false)
aux: ClockTick(100)
aux: Flags(0)
aux: HwCap(0)
aux: HwCap2(0)
aux: PHdr(35184376283200)
aux: PHent(56)
aux: PHnum(10)
aux: Random([83, 133, 20, 165, 6, 211, 83, 230, 210, 35, 42, 218, 8, 210, 151, 177])
```

**Comment by haraldh** [comment]
```console
$ hyperfine "/home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim_old"  "/home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim"
Benchmark #1: /home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim_old
  Time (mean ± σ):      1.553 s ±  0.162 s    [User: 698.5 ms, System: 848.3 ms]
  Range (min … max):    1.362 s …  1.883 s    10 runs
 
Benchmark #2: /home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim
  Time (mean ± σ):      1.558 s ±  0.285 s    [User: 704.0 ms, System: 848.7 ms]
  Range (min … max):    1.301 s …  2.246 s    10 runs
 
Summary
  '/home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim_old' ran
    1.00 ± 0.21 times faster than '/home/haraldh/git/enarx/enarx-keep-sgx-shim/../enarx-keep-sgx/target/debug/enarx-keep-sgx --code ../payload/target/x86_64-unknown-linux-musl/debug/payload --shim target/x86_64-unknown-linux-musl/release/enarx-keep-sgx-shim'

```

File: enarx-keep-sgx-shim/.cargo/config [hunk 1]
```diff
@@ -2,6 +2,5 @@
 target = "x86_64-unknown-linux-musl"
 rustflags = [
     "-C", "relocation-model=pic",
-    "-C", "opt-level=3",
```
Comment (npmccallum) [inline enarx-keep-sgx-shim/.cargo/config:5]: Why don't we want this? Is this automatically included in the release profile?

File: enarx-keep-sgx-shim/src/main.rs [hunk 2]
```diff
@@ -14,6 +14,18 @@ fn panic(_info: &core::panic::PanicInfo) -> ! {
...
+#[no_mangle]
+extern "C" fn _Unwind_Resume() {
+    loop {}
+}
+
+#[cfg(not(test))]
+#[no_mangle]
+pub extern "C" fn rust_eh_personality() {
+    loop {}
+}
```
Comment (npmccallum) [inline enarx-keep-sgx-shim/src/main.rs:27]: Can you give a brief comment as to what these are and why they are here?

File: enarx-keep-sgx-shim/.cargo/config (hunk 1 above)
Comment (haraldh) [reply enarx-keep-sgx-shim/.cargo/config:5]: Yes, this is automatic in `release`... and we don't want this in `debug`

Comment (haraldh) [reply enarx-keep-sgx-shim/.cargo/config:5]: e.g. it removes all `debug_assert` in debug mode

Comment (npmccallum) [reply enarx-keep-sgx-shim/.cargo/config:5]: ACK

File: enarx-keep-sgx-shim/src/main.rs (hunk 2 above)
Comment (npmccallum) [reply enarx-keep-sgx-shim/src/main.rs:27]: (A comment in the code for future readers, that is...)

Comment (haraldh) [reply enarx-keep-sgx-shim/src/main.rs:27]: yes

**Comment by haraldh** [comment]
meh.. sorry, newline `cargo fmt` error...

**Pull request merged**